src/vs/platform/windows/electron-main/windowUtils.ts

```diff
@@ -0,0 +1,16 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import electron from 'electron';
+
+/**
+ * We have some components like `NativeWebContentExtractorService` that create offscreen windows
+ * to extract content from web pages. These windows are not visible to the user and are not
+ * considered part of the main application window. This function filters out those offscreen
+ * windows from the list of all windows.
+ * @returns An array of all BrowserWindow instances that are not offscreen.
+ */
+export function getAllWindowsExcludingOffscreen() {
```

Comment:
@TylerLeonhardt why an extra file? We already have `window.ts` and `windows.ts` in `electron-main` layer, can we not move it there?

Reply:
so that importing didn't cause any side effects. Where do you want it, I don't have a preference.

Reply:
I think in `src/vs/platform/windows/electron-main/windows.ts` is fine.